### PR TITLE
[#4267] Specify objecttypes by their UUID

### LINKED ISSUE #4432: Add error-boundary to the registration modal in the variables tab
### Product versie / Product version

latest (pre 2.7)

### Customer reference

_No response_

### Omschrijf het probleem / Describe the bug

All my hard work gets lost if Objects API is not reachable and I switch to the variables tab to assign variables.

### Stappen om te reproduceren / Steps to reproduce

_No response_

### Verwacht gedrag / Expected behavior

_No response_

### Screen resolution

None

### Device

None

### OS

None

### Browser

_No response_

### PR DESCRIPTION
**Pull request by Viicos**
Partly closes #4267, closes #4432 , replaces https://github.com/open-formulieren/open-forms/pull/4433

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Release management

  - [x] I have labelled the PR as "needs-backport" accordingly

- I have updated the translations assets (you do NOT need to provide translations)

  - [x] Ran `./bin/makemessages_js.sh`
  - [x] Ran `./bin/compilemessages_js.sh`

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

### CODE REVIEW

File: src/openforms/upgrades/upgrade_paths.py [hunk 1]
```diff
@@ -76,6 +76,12 @@ def run_checks(self) -> bool:
 # If your current version falls outside of a supported range, you need to do another
 # upgrade path (first) or there is no upgrade path at all.
 UPGRADE_PATHS = {
+    "2.7": UpgradeConstraint(
+        valid_ranges={
+            VersionRange(minimum="2.6"),
```
Comment (Viicos) [inline src/openforms/upgrades/upgrade_paths.py:81]: Not sure about the minimum version range

**Comment by Viicos** [comment]
Form import is definitely going to be an issue

File: src/openforms/registrations/contrib/objects_api/tests/test_update_payment_status_v2.py [hunk 2]
```diff
@@ -159,6 +169,9 @@ def test_update_payment_status(self):
             public_order_id="TEST-123",
         )
 
+        submission.price = 10.01
+        submission.save()
```
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/objects_api/tests/test_update_payment_status_v2.py:173]: direct assignment is not the right approach here, it creates a broken state - a submission cannot have a price if there's no product with price and payment backend setup.

Instead, provide kwargs to the factory:

```py
            form__payment_backend="demo",
            form__product__price=10.01,
```

File: src/openforms/upgrades/upgrade_paths.py (hunk 1 above)
Comment (sergei-maertens) [reply src/openforms/upgrades/upgrade_paths.py:81]: I have the right one in https://github.com/open-formulieren/open-forms/pull/4446 :)

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
You should be able to check `is_import = self.context.get("is_import", False)` in the `FormRegistrationBackendSerializer` to apply the conversion of URL -> UUID (or raise validation error) to handle importing old forms, in the `validate` method or in `to_internal_value`. That way you can rewrite the data to fit the new format.

File: src/openforms/registrations/contrib/objects_api/submission_registration.py [hunk 3]
```diff
@@ -252,6 +252,11 @@ def save_registration_data(
                             )
                         )
 
+    def build_objecttype_url(self, options: OptionsT) -> str:
+        base_url = options["objects_api_group"].objecttypes_service.api_root
+
+        return f"{base_url}objecttypes/{options['objecttype']}"
```
[the remark this answers is not in the document]
Comment (Viicos) [reply src/openforms/registrations/contrib/objects_api/submission_registration.py:258]: I don't really like the constraint of having to pass a whole `object_type` dict to be able to create an object, can't we just pass the objecttype uuid?

[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/registrations/contrib/objects_api/submission_registration.py:258]: the objects API client does not have the necessary information to build the URL to the objecttypes API

[the remark this answers is not in the document]
Comment (Viicos) [reply src/openforms/registrations/contrib/objects_api/submission_registration.py:258]: Sorry I meant the objecttype url

File: src/openforms/upgrades/upgrade_paths.py (hunk 1 above)
Comment (sergei-maertens) [reply src/openforms/upgrades/upgrade_paths.py:81]: I've merged the squashed migrations PR, so you can rebase on master to add the check!

File: src/openforms/registrations/contrib/objects_api/submission_registration.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/registrations/contrib/objects_api/submission_registration.py:258]: ah yeah, also works for me :) implementation detail

File: src/openforms/upgrades/upgrade_paths.py (hunk 1 above)
Comment (Viicos) [reply src/openforms/upgrades/upgrade_paths.py:81]: thanks

File: src/openforms/registrations/contrib/objects_api/submission_registration.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Viicos) [reply src/openforms/registrations/contrib/objects_api/submission_registration.py:258]: I updated the logic to handle it this way. Lots of tests are updated because I had to add an extra mocked request in v1 tests (because we now hit the objecttypes API during registration).

However, there's an annoying issue with v2 (docker) tests:
- The configured test objecttypes service points to `http://localhost:8001/api/v2/`, and this works when fetching the objecttype URL during registration.
- But the objects API instance has a reference to `http://objecttypes-web:8000/api/v2/"`, as it communicates through the Docker network

[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/registrations/contrib/objects_api/submission_registration.py:258]: you mean in the response data of the objects API, so that making asserts is a bit weird? or are you actually getting network errors?

File: src/openforms/registrations/contrib/objects_api/tests/files/vcr_cassettes/ObjectsAPIBackendV2Tests/ObjectsAPIBackendV2Tests.test_submission_with_file_components.yaml [hunk 4]
```diff
@@ -130,27 +166,25 @@ interactions:
     uri: http://localhost:8002/api/v2/objects
   response:
     body:
-      string: '{"url":"http://localhost:8002/api/v2/objects/7d9d7389-9fea-43bf-a630-84a60eaab373","uuid":"7d9d7389-9fea-43bf-a630-84a60eaab373","type":"http://objecttypes-web:8000/api/v2/objecttypes/527b8408-7421-4808-a744-43ccb7bdaaa2","record":{"index":1,"typeVersion":1,"data":{"single_file":"http://localhost:8003/documenten/api/v1/enkelvoudiginformatieobjecten/34b8ef68-44f3-433c-895c-14b4dd975356","multiple_files":["http://localhost:8003/documenten/api/v1/enkelvoudiginformatieobjecten/2bca92b5-a0c2-4ec3-9361-85f7c13451cb"]},"geometry":null,"startAt":"2024-03-19","endAt":null,"registrationAt":"2024-03-22","correctionFor":null,"correctedBy":null}}'
+      string: '{"detail":"You do not have permission to perform this action."}'
```
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/objects_api/tests/files/vcr_cassettes/ObjectsAPIBackendV2Tests/ObjectsAPIBackendV2Tests.test_submission_with_file_components.yaml:169]: is this how the different hosts names issue manifests?

File: src/openforms/registrations/contrib/objects_api/tests/test_update_payment_status_v2.py [hunk 5]
```diff
@@ -152,13 +162,6 @@ def test_update_payment_status(self):
             "geometry_variable_key": "location",
         }
 
-        SubmissionPaymentFactory.create(
-            submission=submission,
-            status=PaymentStatus.completed,
-            amount=10.01,
-            public_order_id="TEST-123",
-        )
```
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/objects_api/tests/test_update_payment_status_v2.py:160]: why is this removed? the tests is specifically about payment status update, which is triggered when the payment is completed. The submissionpayment record being present *is* needed for a representative state

File: src/openforms/registrations/contrib/objects_api/submission_registration.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/registrations/contrib/objects_api/submission_registration.py:258]: Hrm, so the URL that enters objects-api is `http://localhost:8001/api/v2/objecttypes`, which objects API tries to resolve but can't because the service is defined as `http://objecttypes-web:8000/api/v2/` instead.

I was looking if we can leverage `host.docker.internal`, but that doesn't seem a solution either because it requires us to edit `/etc/hosts` on the host system and that's just too annoying.

However - I think we can do some reverse-proxy trickery in docker-compose - if we put nginx in frontend of object types and objects API, we can make that rewrite `localhost:8000` and `localhost:8001` URLs to a `Host` header of `http://objects-web:8000` and `http://objecttypes-web:8000`. So we can still poke in from outside through localhost, and we still reach the URLs inside.

**Comment by Viicos** [comment]
This is still missing an update in the `TargetPathsInputSerializer`

**Review by sergei-maertens — APPROVED** [review]
